src/plugins/auto/src/plugin.cpp

```diff
@@ -404,8 +404,13 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model_impl(const std::string
         support_devices = filter_device_by_model(support_devices_by_property, model, load_config);
         cloned_model = model->clone();
     } else {
-        // AUTO / MULTI don't support caching explicitly, but can redirect this functionality to actual HW plugin
         LOG_INFO_TAG("compile model with model path");
+        if (work_mode_auto) {
+            cloned_model = get_core()->read_model(model_path, std::string{});
+            support_devices = filter_device_by_model(support_devices_by_property, cloned_model, load_config);
+        } else {
+            auto_s_context->m_model_path = model_path;
+        }
```

Comment:
@wangleis this will degrade model compile latency.
do you see other better solutions? not sure if possible to get stateful info from cache or model file?

Reply:
@wangleis all of the OV HW plugins, including CPU, GPU and NPU, only have one compile model API that only accepts the model object, instead of model path, as the input parameter. The compile latency may not change when  AUTO calls `read_model()` before try to compile model to HW plugins. However, Core has been implemented an virtual compile model API that will returns a model object created by calling `read_model()` API, which means HW plugin can override this API.

- compile model API in CPU plugin: https://github.com/openvinotoolkit/openvino/blob/421eaec1b52d6813035a94a4548ed4c1de0ef374/src/plugins/intel_cpu/src/plugin.h#L18
- compile model API in GPU plugin:https://github.com/openvinotoolkit/openvino/blob/421eaec1b52d6813035a94a4548ed4c1de0ef374/src/plugins/intel_gpu/include/intel_gpu/plugin/plugin.hpp#L51
- compile model API in NPU plugin: https://github.com/openvinotoolkit/openvino/blob/421eaec1b52d6813035a94a4548ed4c1de0ef374/src/plugins/intel_npu/src/plugin/include/plugin.hpp#L36

Reply:
After detailed performance checks during model compile phase, the mem utilization and latency has no obvious changes when AUTO passes loaded model(`read_model()`), instead of model path, to HW plugin via Core. @wangleis @songbell

Reply:
I don't think so, if core.compile_model("test.xml", GPU) with cache enabled can work, why auto cannot benefit?

Reply:
Currently, only test compile model with cache disabled and no degrade happened for AUTO in this situation. Will check the performance change for compile model path with cache enabled soon.

Reply:
No performance gap was observed on 12 different scale models when compiling the model with cache enabled.